### PR TITLE
fix pattern to only match word chars

### PR DESCRIPTION
**Pull request by camdencheek**

### CODE REVIEW

File: lib/kernel/match/match_chunk.ml [hunk 1]
```diff
@@ -54,3 +54,4 @@ let to_chunks source l =
   |> List.sort ~compare:(fun (left : Range.t) right -> Int.compare left.match_start.offset right.match_start.offset)
   |> fold_matches []
   |> range_chunks source
+  |> List.rev
```
Comment (camdencheek) [inline lib/kernel/match/match_chunk.ml:57]: Reverse the chunks so they are in order (side note: can you really not append / pop with ocaml lists??)

File: test/common/test_match_chunk_conversion.ml [hunk 2]
```diff
@@ -14,8 +14,13 @@ d
 e
 baz(qux)
 f
+foo(
+    bar
+    baz
+    qux
+)
```
Comment (camdencheek) [inline test/common/test_match_chunk_conversion.ml:21]: Add a multiline case

File: test/common/test_match_chunk_conversion.ml [hunk 3]
```diff
@@ -27,17 +32,33 @@ g|} in
...
+    "ranges": [
       {
-        "start": { "offset": 14, "line": 4, "column": 9 },
+        "start": { "offset": 19, "line": 7, "column": 1 },
         "end": { "offset": 27, "line": 7, "column": 9 }
       }
     ]
+  },
+  {
+    "content": "foo(\n    bar\n    baz\n    qux\n)",
```
Comment (camdencheek) [inline test/common/test_match_chunk_conversion.ml:55]: yay!

File: test/common/test_match_chunk_conversion.ml [hunk 4]
```diff
@@ -14,8 +14,13 @@ d
...
 baz(qux)
 f
+foo(
+    bar
+    baz
+    qux
+)
 g|} in
-  let template = ":[x](:[y])" in
+  let template = ":[x~\\w+](:[y])" in
```
[the remark this answers is not in the document]
Comment (rvantonder) [reply test/common/test_match_chunk_conversion.ml:23]: `:[[x]]` maybe more readable this is fine as is

File: lib/kernel/match/match_chunk.ml (hunk 1 above)
Comment (rvantonder) [reply lib/kernel/match/match_chunk.ml:57]: you can append with `@`

**Comment by rvantonder** [comment]
Just gonna merge into my branch since it makes progress :-)

File: lib/kernel/match/match_chunk.ml (hunk 1 above)
Comment (camdencheek) [reply lib/kernel/match/match_chunk.ml:57]: Huh. I tried that. It complained and stack overflow was pessimistic

**Pull request merged**